stdlib/system/src/main/java/org/ballerinalang/stdlib/system/utils/SystemUtils.java

```diff
@@ -55,33 +55,30 @@
      * @return Ballerina error object.
      */
     public static ErrorValue getBallerinaError(String reason, Throwable error) {
-        String errorMsg = error != null && error.getMessage() != null ? error.getMessage() : UNKNOWN_MESSAGE;
-        return getBallerinaError(reason, errorMsg);
+        String errorMsg = error != null && error.getMessage() != null ? error.getMessage() : reason;
+        return getBallerinaError(errorMsg);
     }
 
     /**
      * Returns error object for input reason and details.
      * Error type is generic ballerina error type. This utility to construct error object from message.
      *
-     * @param reason  Reason for creating the error object. If the reason is null, value "UNKNOWN" is set by
-     *                default.
      * @param details Java throwable object to capture description of error struct. If throwable object is null,
      *                "Unknown Error" is set to message by default.
      * @return Ballerina error object.
      */
-    public static ErrorValue getBallerinaError(String reason, String details) {
-        MapValue<String, Object> refData = new MapValueImpl<>(BTypes.typeError.detailType);
-        if (reason != null) {
-            reason = SystemConstants.ERROR_REASON_PREFIX + reason;
-        } else {
-            reason = SystemConstants.ERROR_REASON_PREFIX + UNKNOWN_REASON;
-        }
-        if (details != null) {
-            refData.put("message", details);
+    public static ErrorValue getBallerinaError(String details) {
+        return BallerinaErrors.createError(SystemConstants.SYSTEM_ERROR_CODE, populateFilepathErrorRecord(details));
+    }
+
+    private static MapValue populateFilepathErrorRecord(String message) {
```

Comment:
Rename the method to `populateSystemErrorRecord`

Reply:
Resolved.